src/Core/Domain/Carrier/ValueObject/ShippingRange.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Exception\CarrierConstraintException;
+
+/**
+ * Provides shipping range and its prices by zone
+ */
+final class ShippingRange
+{
+    /**
+     * @var int
+     */
+    private $from;
+
+    /**
+     * @var int
+     */
+    private $to;
+
+    /**
+     * @var array
+     */
+    private $pricesByZoneId;
+
+    /**
+     * @param int $from
+     * @param int $to
+     * @param array $pricesByZoneId
```

Comment:
You could typehint something more specific if possible, like `string[]`.

Reply:
Its id => value pairs array, I think typehinting could be missleading, no? I could do like float[], but IMO this way makes it look like the index value is not important.